java/compaction/compaction-job-execution/src/main/java/sleeper/compaction/job/execution/CompactSortedFilesRunner.java

```diff
@@ -79,8 +81,6 @@
  */
 public class CompactSortedFilesRunner {
```

Comment:
The current behaviour in this class seems to be that if there are no jobs at all, it waits for the wait period as many times as the max retrieve attempts that's configured.

It seems a little odd that these two things are coupled, particularly as any exception that's thrown during a job also counts as a time when there were no messages, when there really were.

Maybe we should be able to just set how long we want the task to stay around for when there are no jobs for it to pick up? It could check how much time has actually passed since it finished its last job?

The wait time for the SQS queue can be its own thing to specifically configure how long it waits on SQS.

The maximum consecutive failures could be a separate parameter so that if a task seems to fail processing any jobs no matter what, it'll stop.